packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -160,3 +169,55 @@ describe('product set', () => {
         })
     })
 })
+
+describe.only('Recommended Products', () => {
+    let fetchMock
+    beforeAll(() => {
+        fetchMock = jest.spyOn(global, 'fetch').mockImplementation(async (url) => {
+            const json = url.endsWith('viewed-recently-einstein') ? einsteinRecommendation : {}
+            return new Response(JSON.stringify(json))
+        })
+    })
+    beforeEach(() => {
+        fetchMock.mockClear()
+    })
+    afterAll(() => {
+        fetchMock.mockRestore()
+    })
+    test('Recently Viewed gets updated when navigating between products', async () => {
+        global.server.use(
+            // Use a single product (and not a product set)
+            rest.get('*/products/:productId', (req, res, ctx) => {
+                return res(ctx.json(productsResponse.data[0]))
+            }),
+            rest.get('*/products', (req, res, ctx) => {
+                return res(ctx.json({}))
+            })
+        )
+        const user = userEvent.setup({
+            advanceTimers: jest.advanceTimersByTime
+        })
+        renderWithProviders(<MockedComponent />)
+
+        let done = false
+        setTimeout(() => (done = true), 150)
+        await waitFor(() => expect(done).toBeTruthy())
+
+        await waitFor(() => {
+            // Main product has loaded
+            expect(screen.getAllByText(/Long Sleeve Crew Neck/)).toHaveLength(2)
+            // Recently Viewed has loaded
+            expect(screen.getByText(/Summer Bomber Jacket/)).toBeInTheDocument()
+        })
+
+        // We requested Recently Viewed products on the first page load, but we
+        // only want to check against the second page load
+        fetchMock.mockClear()
+        await user.click(screen.getByText(/Summer Bomber Jacket/))
+
+        expect(fetchMock).toHaveBeenCalledWith(
+            expect.stringMatching(/viewed-recently-einstein$/),
+            expect.any(Object)
+        )
```

Comment:
This is currently failing. I think I need to change the mock API responses to return a different product for the second call, but I haven't done that yet.